index.js

```diff
@@ -37,26 +38,35 @@ function fixOrientation (url, opts, fn) {
   var s = size[buf.type](buf);
   var max = Math.max(s.width, s.height);
   var half = max / 2;
-  var dir = { 6: 1, 8: -1 }[tags.Orientation];
+  var rotateDegrees = { 3: 180, 6: 90, 8: -90 }[tags.Orientation];
 
   var canvas = document.createElement('canvas');
   var ctx = canvas.getContext('2d');
   canvas.width = canvas.height = max;
 
-  rotate(ctx, { x: half, y: half, degrees: dir * 90 });
+  rotate(ctx, { x: half, y: half, degrees: rotateDegrees });
 
   urlToImage(url, function (img) {
-    if (dir == 1) {
+    if (6 == tags.Orientation || (tags.Orientation == 3 && s.height < s.width)) {
```

Comment:
why do we need this condition `s.height < s.width`?

Reply:
If you don't, this one will work:
https://github.com/ianare/exif-samples/blob/master/jpg/orientation/landscape_3.jpg
but this one : 
https://github.com/ianare/exif-samples/blob/master/jpg/orientation/portrait_3.jpg
will cut like this:
https://i.imgur.com/81wpG4N.jpg

Reply:
hmm so we will support the `3` tag but only for certain dimensions? I'd rather support all of those images or none, otherwise you never know what you'll get, right?

Reply:
maybe we can move the image in the canvas before or after rotating?

Reply:
It does work for both wider or higher dimensions. (tested with theses samples : https://github.com/ianare/exif-samples/tree/master/jpg/orientation ) 

I can check how to do it without dimension flipping in all cases if I find the time, but it was quicker to reuse the existing paths for our implementation.